jetson/ra_kinematics/main.cpp

```diff
@@ -104,36 +106,69 @@ int main() {
+    json geom = read_json_from_file(get_mrover_arm_geom(science));
 
     lcm::LCM lcmObject;
-    MRoverArm robot_arm(geom, lcmObject);
-
-    lcmHandlers handler(&robot_arm);
-
-    lcmObject.subscribe( "/ra_ik_cmd", &lcmHandlers::armPositionCallback, &handler );
-    lcmObject.subscribe( "/target_orientation" , &lcmHandlers::executeCallback, &handler );
-    lcmObject.subscribe( "/motion_execute", &lcmHandlers::motionExecuteCallback, &handler );
-    lcmObject.subscribe( "/simulation_mode", &lcmHandlers::simModeCallback, &handler );
-    lcmObject.subscribe( "/arm_control_state", &lcmHandlers::armControlCallback, &handler );
-    lcmObject.subscribe( "/use_orientation", &lcmHandlers::useOrientationCallback, &handler );
-    lcmObject.subscribe( "/locked_joints", &lcmHandlers::lockJointsCallback, &handler );
-    lcmObject.subscribe( "/zero_position", &lcmHandlers::zeroPositionCallback, &handler );
-    lcmObject.subscribe( "/arm_adjustments", &lcmHandlers::armAdjustCallback, &handler );
-    lcmObject.subscribe( "/arm_preset", &lcmHandlers::armPresetCallback, &handler );
-
-    Signal signal;
-    lcmObject.publish("/ik_reset", &signal);
-    
-    std::thread execute_spline(&MRoverArm::execute_spline, &robot_arm);
-    std::thread send_arm_position(&MRoverArm::encoder_angles_sender, &robot_arm);
 
-    while( lcmObject.handle() == 0 ) {
-        // run kinematics
+    if (science) {
+        ScienceArm robot_arm(geom, lcmObject);
+
+        lcmHandlers handler(&robot_arm);
+
+        lcmObject.subscribe( "/sa_position", &lcmHandlers::armPositionCallback, &handler );
+        lcmObject.subscribe( "/motion_execute", &lcmHandlers::motionExecuteCallback, &handler );
+        lcmObject.subscribe( "/simulation_mode", &lcmHandlers::simModeCallback, &handler );
+        lcmObject.subscribe( "/arm_control_state", &lcmHandlers::armControlCallback, &handler );
+        lcmObject.subscribe( "/locked_joints", &lcmHandlers::lockJointsCallback, &handler );
+        lcmObject.subscribe( "/zero_position", &lcmHandlers::zeroPositionCallback, &handler );
+        lcmObject.subscribe( "/arm_adjustments", &lcmHandlers::armAdjustCallback, &handler );
+        lcmObject.subscribe( "/arm_preset", &lcmHandlers::armPresetCallback, &handler );
+
+        Signal signal;
+        lcmObject.publish("/ik_reset", &signal);
+
+        std::thread execute_spline(&MRoverArm::execute_spline, &robot_arm);
+        std::thread send_arm_position(&MRoverArm::encoder_angles_sender, &robot_arm);
+
+        while( lcmObject.handle() == 0 ) {
+            // run kinematics
+        }
+
+        execute_spline.join();
+        send_arm_position.join();
+
+        return 0;
     }
+    else {
+        StandardArm robot_arm(geom, lcmObject);
+
+        lcmHandlers handler(&robot_arm);
+
+        lcmObject.subscribe( "/ra_position", &lcmHandlers::armPositionCallback, &handler );
+        lcmObject.subscribe( "/target_orientation" , &lcmHandlers::executeCallback, &handler );
```

Comment:
Does it matter that there's a motion execute callback for the standard arm but not the science arm?

Reply:
That channel is for IK which the science arm won't be using.